Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -54,16 +54,26 @@ extension BrowserViewController {
       return
     }
     
+    var alertDetails = String(format: Strings.openExternalAppURLMessage, url.relativeString)
+    
     // We do not want certain schemes to be opened externally when called from subframes.
-    if ["tel", "sms"].contains(url.scheme) && !isMainFrame {
-      return
+    // And tel / sms dialog should not be shown for non-active tabs #6687
+    if ["tel", "sms"].contains(url.scheme) {
+      if !isMainFrame || tab?.url?.host != topToolbar.currentURL?.host {
```

Comment:
Do we have to compare `host` instead of entire url? what if user has two xxx.com tabs, one of which is not active?

Reply:
No we should not, here we are checking if the website is trying to do some trickery like doing a redirection to a totally new domain. User can have multiple tabs for xxx.com for instance this doesnt change the fact we are comparing tab url host with omnibox url host.

Reply:
thanks for explaining